src/MongoDB.Driver.Core/Core/Operations/AggregateOperation.cs

```diff
@@ -253,33 +264,49 @@ public IBsonSerializer<TResult> ResultSerializer
         public IAsyncCursor<TResult> Execute(IReadBinding binding, CancellationToken cancellationToken)
         {
             Ensure.IsNotNull(binding, nameof(binding));
+
+            using (var context = RetryableReadContext.Create(binding, _retryRequested, cancellationToken))
```

Comment:
We have to go ahead an create the RetryableReadContext here because we are going to base some decisions on the server version of the channel that the RetryableReadContext ends up with.

And then we want to execute the ReadCommand on that same channel, which we do by calling the overload of Execute that takes a context instead of a binding.